`TestDatabaseResource ` flakiness (timeouts)

## Failure

#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/4514436827/jobs/7950522343

#### Relevant snippet

```
panic: test timed out after 10m0s
running tests:
	TestDatabaseResource (9m29s)
```

```
goroutine 22579 [semacquire, 9 minutes]:
sync.runtime_Semacquire(0xc003e26ce8?)
	/opt/go/src/runtime/sema.go:62 +0x27
sync.(*WaitGroup).Wait(0xc003e26ce0)
	/opt/go/src/sync/waitgroup.go:116 +0xa5
github.com/gravitational/teleport/lib/service.(*LocalSupervisor).Wait(0xc001d22a80)
	/__w/teleport/teleport/lib/service/supervisor.go:322 +0x85
github.com/gravitational/teleport/tool/tctl/common.makeAndRunTestAuthServer.func1()
	/__w/teleport/teleport/tool/tctl/common/helpers_test.go:239 +0xa5
testing.(*common).Cleanup.func1()
	/opt/go/src/testing/testing.go:1150 +0x194
testing.(*common).runCleanup(0xc002e7e9c0, 0x0)
	/opt/go/src/testing/testing.go:1328 +0x1ea
testing.tRunner.func2()
	/opt/go/src/testing/testing.go:1570 +0x53
testing.tRunner(0xc002e7e9c0, 0x92155e0)
	/opt/go/src/testing/testing.go:1582 +0x255
created by testing.(*T).Run
	/opt/go/src/testing/testing.go:1629 +0x806
```

```
goroutine 23485 [select, 9 minutes]:
github.com/gravitational/teleport/lib/multiplexer.(*Listener).Accept(0xc0026c4060)
	/__w/teleport/teleport/lib/multiplexer/wrappers.go:135 +0x113
github.com/gravitational/teleport/lib/srv/alpnproxy.(*ListenerMuxWrapper).startAcceptingConnectionServiceListener(0xc002e04a00)
	/__w/teleport/teleport/lib/srv/alpnproxy/listener.go:98 +0xa2
created by github.com/gravitational/teleport/lib/srv/alpnproxy.NewMuxListenerWrapper
	/__w/teleport/teleport/lib/srv/alpnproxy/listener.go:56 +0x27b
```

--- a/lib/service/service.go
+++ b/lib/service/service.go
@@ -3037,9 +3037,7 @@ func (l *proxyListeners) Close() {
 	if l.kube != nil {
 		l.kube.Close()
 	}
-	if !l.db.Empty() {
-		l.db.Close()
-	}
+	l.db.Close()
 	if l.alpn != nil {
 		l.alpn.Close()
 	}

--- a/lib/service/service.go
+++ b/lib/service/service.go
@@ -3011,6 +3011,55 @@ type proxyListeners struct {
 	minimalTLS       *multiplexer.WebListener
 }
 
+// Close closes all proxy listeners.
+func (l *proxyListeners) Close() {
+	if l.mux != nil {
+		l.mux.Close()
+	}
+	if l.sshMux != nil {
+		l.sshMux.Close()
+	}
+	if l.tls != nil {
+		l.tls.Close()
+	}
+	if l.ssh != nil {
+		l.ssh.Close()
+	}
+	if l.sshGRPC != nil {
+		l.sshGRPC.Close()
+	}
+	if l.web != nil {
+		l.web.Close()
+	}
+	if l.reverseTunnel != nil {
+		l.reverseTunnel.Close()
+	}
+	if l.kube != nil {
+		l.kube.Close()
+	}
+	if !l.db.Empty() {
+		l.db.Close()
+	}
+	if l.alpn != nil {
+		l.alpn.Close()
+	}
+	if l.proxyPeer != nil {
+		l.proxyPeer.Close()
+	}
+	if l.grpcPublic != nil {
+		l.grpcPublic.Close()
+	}
+	if l.grpcMTLS != nil {
+		l.grpcMTLS.Close()
+	}
+	if l.reverseTunnelMux != nil {
+		l.reverseTunnelMux.Close()
+	}
+	if l.minimalTLS != nil {
+		l.minimalTLS.Close()
+	}
+}
+
 // dbListeners groups database access listeners.
 type dbListeners struct {
 	// postgres serves Postgres clients.
@@ -3041,46 +3090,6 @@ func (l *dbListeners) Close() {
 	}
 	if l.mongo != nil {
 		l.mongo.Close()
-	}
-}
-
-// Close closes all proxy listeners.
-func (l *proxyListeners) Close() {
-	if l.mux != nil {
-		l.mux.Close()
-	}
-	if l.sshMux != nil {
-		l.sshMux.Close()
-	}
-	if l.tls != nil {
-		l.tls.Close()
-	}
-	if l.web != nil {
-		l.web.Close()
-	}
-	if l.reverseTunnel != nil {
-		l.reverseTunnel.Close()
-	}
-	if l.kube != nil {
-		l.kube.Close()
-	}
-	if !l.db.Empty() {
-		l.db.Close()
-	}
-	if l.grpcPublic != nil {
-		l.grpcPublic.Close()
-	}
-	if l.grpcMTLS != nil {
-		l.grpcMTLS.Close()
-	}
-	if l.proxyPeer != nil {
-		l.proxyPeer.Close()
-	}
-	if l.reverseTunnelMux != nil {
-		l.reverseTunnelMux.Close()
-	}
-	if l.minimalTLS != nil {
-		l.minimalTLS.Close()
 	}
 }
 

